draft-ietf-detnet-dp-sol-ip-02.xml

```diff
@@ -1120,6 +1124,121 @@ DetNet                    |L2/SbN|          |L2/SbN|
 
     <!-- ===================================================================== -->
 
+    <section anchor="ip-over-mpls" title="DetNet IP over DetNet MPLS Operations">
+      <t>
+        This section defines how IP encapsulated DetNet flows are
+        carried over a DetNet MPLS data plane as defined in <xref
+        target="I-D.ietf-detnet-dp-sol-mpls"/>.  The basic encapsulation
+        approach is to treat a DetNet IP flow as an app flow from the
+        DetNet MPLS app perspective. The corresponding example DetNet
+        Sub-Network format is shown in <xref
+        target="fig_dn_ip_mpls_sn_ex"/>.
+      </t>
+      <figure title="Example DetNet IP over MPLS Sub-Network Formats"
+              anchor="fig_dn_ip_mpls_sn_ex">
+        <artwork align="center"><![CDATA[
+           /->     +------+  +------+  +------+
+           |       |  X   |  |  X   |  |  X   |  IP App-Flow
+MPLS       |       +------+  +------+  +------+
+         <-+       |NProto|  |NProto|  |NProto|
+App-Flow   |       +------+  +------+  +------+
+           |       |  IP  |  |  IP  |  |  IP  |
+           \-> +---+======+--+======+--+======+-----+
+DetNet-MPLS        | d-CW |  | d-CW |  | d-CW |
+                   +------+  +------+  +------+
+                   |Labels|  |Labels|  |Labels|
+               +---+======+--+======+--+======+-----+
+Sub-Network        |  L2  |  | TSN  |  | UDP  |
+                   +------+  +------+  +------+
+                                       |  IP  |
+                                       +------+
+                                       |  L2  |
+                                       +------+
+    ]]>
+        </artwork>
+      </figure>
+      <t>
+        In the figure, IP App-Flow indicates the application flow
+        carried in by the DetNet IP data plane.  IP and NProto indicate
```

Comment:
We need to sort out flow specific terminology. My understanding:
- App-flow: native format of a DetNet-flow. Examples: TSN Stream (Layer-2), L2 flow (Layer-2), IP flow (Layer-3). Note: L2 flow is anything what is L2 encapsulated. It can be an IP flow together with its Ethernet header if the DetNet domain serves it as a L2 frame.
- DetNet flow: is the "App-flow plus its DetNet encapsulation". DetNet encapsulation can be a zero-encapsulation, i.e., no additional header (e.g., IP flow sent over an IP DetNet network, then the "IP flow" and the "IP DetNet flow" are exactly the same over the wire). DetNet encapsulation is non-zero in case of MPLS DetNet networks, i.e., dCW and S-label are added to the App-flow.
Note: F-label(s) are not part of the DetNet flow by definition.
There are two types of DetNet flows:
1, IP DetNet flow: contains IP packets
2, MPLS DetNet flow: contains labeled packets (dCW+S-label)

Is that OK? I have found the term like "MPLS App-flow" confusing.